build/common.sh

```diff
@@ -574,11 +574,17 @@ function kube::build::run_build_command_ex() {
     --env "KUBE_BUILD_WITH_COVERAGE=${KUBE_BUILD_WITH_COVERAGE:-}"
     --env "KUBE_BUILD_PLATFORMS=${KUBE_BUILD_PLATFORMS:-}"
     --env "GOFLAGS=${GOFLAGS:-}"
-    --env "GOLDFLAGS=${GOLDFLAGS:-}"
     --env "GOGCFLAGS=${GOGCFLAGS:-}"
     --env "SOURCE_DATE_EPOCH=${SOURCE_DATE_EPOCH:-}"
   )
 
+  # use GOLDFLAGS only if it is set explicitly.
+  if [[ -v GOLDFLAGS ]]; then
```

Comment:
If for some reason we do wind up needing to support older bash it's possible to instead do:
```suggestion
  if [[ ! -z ${GOLDFLAGS+unset} ]]; then
```
(`${FOO+x}` will evaluate to nothing if FOO is set, or else x, so if the length is zero then it wasn't set)

xref: https://github.com/kubernetes/kubernetes/issues/98294

I don't think we should need to though.